sdk/core/azure-core-management/src/main/java/com/azure/core/management/provider/DelayProvider.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.management.provider;
+
+import java.time.Duration;
+
+/**
+ * The DelayProvider to help thread sleep.
+ */
+public interface DelayProvider {
+
+    /**
+     * Thread sleep.
+     *
+     * @param milliseconds time for sleep
+     */
+    void sleep(int milliseconds);
+
+    /**
+     * Get the retry timeout for long running operation.
+     *
+     * @return the timeout value.
+     */
+    Duration getLroRetryTimeout();
```

Comment:
This one might be removed. Directly we only need `getDelayDuration`. This could be done by a helper method elsewhere.